src/librbd/internal.cc

```diff
@@ -1351,9 +1351,7 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
     int r = ictx->state->open(true);
     if (r < 0) {
       ldout(cct, 2) << "error opening image: " << cpp_strerror(-r) << dendl;
-      if (r != -ENOENT) {
```

Comment:
@runsisi On -ENOENT from image open, just try to retrieve the image id from the directory. If successful, continue -- otherwise, fail with the error code from the directory lookup.